rumydata/rules/header.py

```diff
@@ -9,23 +9,34 @@
 from typing import List
 
 from rumydata._base import _BaseRule
+from rumydata import field
 
 # this named tuple is here to allow for setting the default argument without
 # needing to import the Layout class, which results in a circular import
-_default_thing = namedtuple('DefaultDict', ['layout', 'header_mode'])
+_default_thing = namedtuple('DefaultDict', ['layout', 'header_mode', 'empty_cols_ok'])
 
 
 class Rule(_BaseRule):
     """ Header Rule """
 
-    _default_args = (_default_thing({}, 'exact'),)
+    _default_args = (_default_thing({}, 'exact', False),)
 
     def __init__(self, columns):
         super().__init__()
         self.header_mode = columns.header_mode
         self.definition = columns.layout
+        self.header_mode = columns.header_mode
+        self.layout = columns
+        self.empty_cols_ok = columns.empty_cols_ok
 
     def _prepare(self, data: List[str]) -> tuple:
```

Comment:
If `empty_cols_ok` is set we'll restructure the layout's _definition_, and modify the row data to account for the empties, making them 'match' and therefore pass our header checks